core/shared/src/main/scala/fs2/internal/Algebra.scala

```diff
@@ -193,6 +228,7 @@ private[fs2] object Algebra {
         case o: Output[F,O2] => Output[G,O2](o.values)
         case Run(values) => Run[G,O2,X](values)
         case Eval(value) => Eval[G,O2,X](u(value))
+        case un:Uncons[F,x,O2] => Uncons[G,x,O2](FreeC.suspend(un.s.translate(algFtoG)), un.chunkSize, un.maxSteps).asInstanceOf[Algebra[G,O2,X]]
```

Comment:
Why is suspend needed here?

Reply:
I had problems with SoE and transform. Perhaps with delaying viewL, this could be removed too.